### PR TITLE
Place page small fixes

### PR DESCRIPTION
**Pull request by beets**
* Update ranking labels so the rank direction is clearer
* use geoOverlaps for child places, to get some zip codes which might go outside of the containing place's bounds

### CODE REVIEW

**Comment by beets** [comment]
image

File: server/routes/api/place.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def child_fetch(dcid):
         '/node/property-values',
         {
             'dcids': [dcid],
-            'property': 'containedInPlace',
+            'property': 'geoOverlaps',
```
Comment (shifucun) [inline server/routes/api/place.py:142]: might also change the function name "child_fetch" to overlap_geo_fetch?

**Comment by beets** [comment]
Sorry this morphed into a bigger fix... now also merges equivalent place types, filters geonames and places with 0 population. Also fixes bug where we always used 2018 for the population, which doesn't exist for international geos.

ptal

**Comment by beets** [comment]
Some results against prod mixer:
image

image

image

I didn't create equivalent groupings for Eurostat and AA because there were some weird mappings (e.g. England is AA1 but Scotland, Wales, and Northern Ireland are AA1)

File: server/routes/api/place.py [hunk 2]
```diff
@@ -16,13 +16,21 @@
...
+from flask import Blueprint, jsonify, request, Response
 
 from cache import cache
 from services.datacommons import fetch_data
-from routes.api.stats import get_stats_wrapper
+import routes.api.stats as stats_api
 
-WANTED_PLACE_TYPES = [
+# Place types to keep for list of child places, keyed by parent place type.
+WANTED_PLACE_TYPES = {}
```
Comment (shifucun) [inline server/routes/api/place.py:26]: can merge this into one statement?

WANTED_PLACE_TYPES = {
  'Country': [],
  'State': [],
  'County': []
}

File: server/routes/api/place.py [hunk 3]
```diff
@@ -68,7 +93,7 @@ def name():
     for dcid in dcids:
         values = response[dcid].get('out')
         result[dcid] = values[0]['value'] if values else ''
-    return json.dumps(result)
+    return Response(json.dumps(result), 200, mimetype='application/json')
```
Comment (shifucun) [inline server/routes/api/place.py:96]: curious what is the difference with the bare json.dump

File: server/routes/api/place.py [hunk 4]
```diff
@@ -114,36 +141,54 @@ def child(dcid):
...
+            if place_type in wanted_types and place_pop > 0:
                 result[place_type].append({
                     'name': place.get('name', place['dcid']),
                     'dcid': place['dcid'],
-                    'pop': pop.get(place['dcid'], 0)
+                    'pop': place_pop,
                 })
+
     # Filter equivalent place types
     for (preferred, equivalent) in EQUIVALENT_PLACE_TYPES.items():
```
Comment (shifucun) [inline server/routes/api/place.py:182]: Please add some comments for the logic.

File: server/routes/api/stats.py [hunk 5]
```diff
@@ -26,6 +26,27 @@
 
 # TODO(shifucun): add unittest for this module
 
+def get_stats_latest(dcid_str, stats_var):
```
Comment (shifucun) [inline server/routes/api/stats.py:29]: nice change!

File: server/routes/api/place.py [hunk 6]
```diff
@@ -16,13 +16,21 @@
...
+import routes.api.stats as stats_api
 
-WANTED_PLACE_TYPES = [
+# Place types to keep for list of child places, keyed by parent place type.
+WANTED_PLACE_TYPES = {}
+WANTED_PLACE_TYPES['Country'] = [
+    "State", "Province", "County", "EurostatNUTS1", "EurostatNUTS2",
+    "AdministrativeArea1"
+]
+WANTED_PLACE_TYPES['State'] = ["Province", "County"]
```
Comment (pradh) [inline server/routes/api/place.py:31]: I want to spend a few minutes to lay out the choices here.

1. A place may have equivalent place types, which one should we pick from among them?
      - AA1/State/Province: pick AA1?   (this makes it general across countries)
      - County/AA2: pick AA2?
      - City/Village/Borough/Town: pick City?
      - Neighborhood
      - Zip Codes
      Note1: EurostatNUTS classification is separate from above, but may overlap (e.g. NUTS2 might AA1, NUTS3 might be AA2 for some geos)
      Note2: There are two general types "AdministrativeArea" and "Place" (this is often a City equivalent)

2. A place may have multiple types, some equivalent and some not. Do we pick one type for a place, or multiple (one for each equivalent set)?
      - if we pick multiple then the same node may occur twice in the children listing, which is odd.
      - were we to pick one place, the preference list should be in kPlaceTypesWithPopObs.

3. For a given parent type, which child types to pick/drop?
      - Looks like this is what WANTED_PLACE_TYPES does. But I suspect the key should consider all State-equivalent types, County-equivalent types?

4. For a parent node, do we group geos by "equivalent" types?
      - If one place had a type AA2 and another had County we should probably group them.

File: server/routes/api/place.py [hunk 7]
```diff
@@ -43,16 +51,33 @@
...
+
+
+@cache.memoize(timeout=3600 * 24)  # Cache for one day.
+def get_place_type(place_dcid):
+    place_types = get_property_value(place_dcid, 'typeOf')
+    # We prefer to use specific type like "State", "County" over
+    # "AdministrativeArea"
+    chosen_type = None
+    for place_type in place_types:
+        if not chosen_type or chosen_type.startswith('AdministrativeArea'):
```
Comment (pradh) [inline server/routes/api/place.py:76]: We want to skip Place too/

File: server/routes/api/place.py [hunk 8]
```diff
@@ -16,13 +16,21 @@
...
 from cache import cache
 from services.datacommons import fetch_data
-from routes.api.stats import get_stats_wrapper
+import routes.api.stats as stats_api
 
-WANTED_PLACE_TYPES = [
+# Place types to keep for list of child places, keyed by parent place type.
+WANTED_PLACE_TYPES = {}
+WANTED_PLACE_TYPES['Country'] = [
+    "State", "Province", "County", "EurostatNUTS1", "EurostatNUTS2",
```
Comment (pradh) [inline server/routes/api/place.py:28]: Why County?

File: server/routes/api/place.py (hunk 6 above)
Comment (beets) [reply server/routes/api/place.py:31]: Chatted offline, what we'll do is:

1. ignore AAx for non US, State/County/City for US. Eurostat* will stay separate

2. we'll show the place multiple times for non-equivalent types

3. pick a preferred type from an equivalent set, before key'ing off the list. so the list should have all equivalent place types
Equivalent types for US (geoId/... or zip/...)
State == AA1
County == AA2
City == Town / Village / Burrough
Everywhere else, pick AAx

4. For non-US geos, ignore Place and AA if parent AA2 onwards - group under Place

File: server/routes/api/place.py (hunk 8 above)
Comment (pradh) [inline server/routes/api/place.py:28]: We can drop Province I think

File: server/routes/api/place.py (hunk 6 above)
Comment (pradh) [reply server/routes/api/place.py:31]: Thanks for the summary.

4. To confirm, we want to ignore child AA + Place for parent AA1 and Country, but keep for parent AA2, AA3, ...?

**Review by pradh — DISMISSED** [review]
This is improvement from before. Please feel free to defer the discussed changes to future PR.

File: server/routes/api/place.py (hunk 2 above)
Comment (beets) [reply server/routes/api/place.py]: done

File: server/routes/api/place.py (hunk 3 above)
Comment (beets) [reply server/routes/api/place.py:97]: the browser recognizes the content as json, and displays nice formatting and folding features :) it was helpful for debugging.

File: server/routes/api/place.py (hunk 4 above)
Comment (beets) [reply server/routes/api/place.py:183]: added

File: server/routes/api/stats.py (hunk 5 above)
Comment (beets) [reply server/routes/api/stats.py:29]: thanks, we were doing this too often :)

File: server/routes/api/place.py (hunk 6 above)
Comment (beets) [reply server/routes/api/place.py]: will do this in a follow up!

File: server/routes/api/place.py (hunk 7 above)
Comment (beets) [reply server/routes/api/place.py]: done

File: server/routes/api/place.py (hunk 8 above)
Comment (beets) [reply server/routes/api/place.py]: good catch. removed..

Comment (beets) [reply server/routes/api/place.py]: done

File: server/routes/api/place.py (hunk 3 above)
Comment (shifucun) [reply server/routes/api/place.py:97]: nice, can gradually move the rest of the reponse

**Pull request merged**